CONTRIBUTING_JS.md

```diff
@@ -299,23 +267,23 @@ const API = require('ipfs-api')
 
 There are two possibilities: either it didn’t work out for us, or we don’t know about it. If you think we might have missed it please tell us, but please believe us if we say we tried and it didn’t work for us.
 
-#### Why not use simple npm scripts instead of gulp?
-
-Gulp is not a hard dependency. It’s just a simple way to structure our tasks at the moment. Usually projects only depend on the aegir binaries completely hiding the fact that we are using gulp under the hood. So we are free if we want to switch it out without any issues. We all enjoy npm scripts, and are using them to call the aegir binaries, but there is no nice way of sharing them yet.
-
 #### Where are all the semicolons?
 
 Our linting rules are compatible with [standard](https://github.com/feross/standard), which has many examples on documentation on this. Please go there and read it if you're still curious.
 
-#### Why are you bothering with ES2015 and all this build setup?
+#### Why are you bothering with TypeScript and all this build setup?
+
+We have a large number of modules that integrate together and we've found that the only way to do this at scale without increasing friction to burning point is to have a type system.
+
+At the time of writing a type system based on TypeScript [may be coming to JavaScript](https://github.com/tc39/proposal-type-annotations) so we're only a little ahead of the curve on that front.
 
-We want to see the web move forward, and some of us enjoy writing their JavaScript with things like `const` and arrow functions.
+We've also found that having types means there's less magic in our codebases since it becomes harder to do use the extreme ends of JavaScript's flexibility which then makes everything easier to follow, lowering cognitive overhead and maintenance burden.
 
-#### Do I have to use ES2015, Babel and aegir in my project?
+#### Do I have to use ESM/TypeScript and/or aegir in my project?
 
-No.
+No, but you will find yourself solving problems that have already been solved.
 
-#### Do I have to bundle everything with webpack?
+#### Do I have to bundle everything with esbuild?
 
 No. But other people might ask you to at some point, so it may be better to be prepared.
```

Comment:
aegir+esbuild has proven troublesome for use with front-end projects and component-lib packages (e.g. ipld-explorer-components).

it would be really nice to have some examples and/or direction so we can say aligned with @ipfs/helia-dev best practices.

Reply:
I don't think we have any right now because aegir is mostly used for library-style modules - maybe you could write some?

Would it mean choosing a frontend framework for the user?  Sounds fraught with peril and/or almost immediate obsolescence.

Reply:
Not choosing a frontend framework at all, but choosing a standard bundler and config. We already have standard for style, mocha+playwright for testing, and esbuild for isomorphic non-frontend libs.

Choosing webpack/vite/rollup/snowpack or similar for client-side libraries and webpages, advising against react-scripts, would benefit us in the long run.

It would be great to bundle this into aegir, but maybe an extension of aegir would be a better call.